Assets/Scripts/Actors/ActorPhysics.cs

```diff
@@ -22,6 +22,10 @@ public sealed class ActorPhysics : ActorComponent
 	PhysicsStateType _currentStateType = PhysicsStateType.Jumping;
 	PhysicsState _currentState = new DefaultState();
 	Dictionary<PhysicsStateType, PhysicsState> _stateMap = new Dictionary<PhysicsStateType, PhysicsState>();
+
+	public delegate void StateCallback( PhysicsStateType physicsStateType );
```

Comment:
The reason I had them separate is so different things can subscribe only to one or the other. For example, the camera doesn't need to know about which state is being left, only which is being entered.